skillful_network_server/src/test/java/fr/uca/cdr/skillful_network/services/user/FollowStateTrackerServiceTest.java

```diff
@@ -0,0 +1,171 @@
+import org.hamcrest.core.IsEqual;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.*;
+
+@ActiveProfiles("test")
+@RunWith(SpringRunner.class)
+public class FollowStateTrackerServiceTest {
+
+    @TestConfiguration
+    static class TestConfigurationServiceTest {
+        @Bean
+        public FollowStateTrackerService followStateTrackerService() {
+            return new FollowStateTrackerServiceImpl();
+        }
+    }
+
+    @Autowired
+    private FollowStateTrackerService followStateTrackerService;
+
+    @MockBean
+    private AuthenticationService authenticationService;
+
+    @MockBean
+    private UserService userService;
+
+    @MockBean
+    private FollowStateTrackerRepository followStateTrackerRepository;
+
+    @MockBean
+    private NotificationRepository notificationRepository;
+
+
+    @Before
+    public void setUp(){
+
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    // Follower methods
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+
+    /*public Optional<List<FollowStateTracker>> getFSTByFollowerID(Long followerID) {
+        return Optional.of( fstRepository.findAllByFollower(userService.getById(followerID)) );
+    }*/
+
+    @Test
+    public void testGetALLFSTByFollower(){
+        final Boolean[] passingMethods = {false, false};
+        final int PASSING_FIND_ALL= 0;
+        final int  GET_BY_ID = 1;
+        final User userFollowed = new User();
+        final User userFollower = new User();
+        final List<FollowStateTracker> fstList = new ArrayList<>();
+        final Set<FollowStateTracker> fstSet = new HashSet<>();
+
+        FollowStateTracker fst = new FollowStateTracker(userFollowed,userFollower);
+        fstList.add(fst);
+        fstSet.add(fst);
+
+        userFollower.setFollowerSet(fstSet);
+        userFollowed.setFollowableSet(fstSet);
+
+        Mockito.when(followStateTrackerRepository.findAllByFollower(userFollower))
+                .thenReturn(fstList);
+        Mockito.when(userService.getById(userFollower.getId())).thenReturn(userFollower);
```

Comment:
These lines aren't necessary, are they?
I think the Mockito.DoAnswer methods just below do the same job but I'm not 100% sure.
WDYT ? @danglotb @pfouque06

Reply:
Imo, you are right. It worth a try. If you comment these lines and rerun the test, do you observe the same behavior ? Does the test still pass ? Maybe your assertions aren’t strong enough but it is no big deal here.
I agree that we can remove them (a priori)

Reply:
If I comment these lines, I observe the same behavior and the test still pass. So I think I can remove these lines

Reply:
Concerning assertions, what do you suggest to add ? 
I was thinking to add : 
- the list's size is equal to 1;
- Maybe add names to the users and compare that they are the same at the end in the method response ?